explainers/hidden-content-explainer.md

```diff
@@ -71,196 +75,152 @@ currently possible.
-3. The beforematch event is fired on the element containing "lorem ipsum."
+3. The browser removes the `hidden=until-found` attribute on the element
+   containing the text "lorem ipsum".
 4. User may click on the section to collapse it again.
 
-In this example, most of the content of the page is hidden in collapsed sections.
-It uses the `content-visibility: hidden-matchable` CSS property to hide the
-content while letting it be searchable by find-in-page. When a match is found,
-and `beforematch` event is fired, we expand the section by removing the
-collapsed class.
+In this example, most of the content of the page is hidden in collapsed
+sections.  It uses the `hidden=until-found` HTML attribute to hide the content
+while letting it be searchable by find-in-page. When a match is found, the
+browser automatically expands the section by removing the `hidden` attribute.
 
 Note that the net effect of this is that the user is able to find matches in
-collapsed sections which are automatically expanded. This is possible due to
-both hidden-matchable content and the `beforematch` event.
+collapsed sections which are automatically expanded.
 
 The same effect occurs when scroll-to-text navigation targets text in a hidden
-section. For instance, navigating to
-`example.com/page.html#:~:text=lorem` would expand the introduction section by
-observing the `beforematch` event for that section as a result of the
+section. For instance, navigating to `example.com/page.html#:~:text=lorem` would
+make the browser expand the introduction section as a result of the
 scroll-to-text fragment navigation match.
 
 One possible real-world application of this is the collapsed sections on mobile
-wikipedia pages. find-in-page and scroll-to-text currently can't find text inside
-of the collapsed sections, but with beforematch and content-visibility they
-could be searchable and automatically expanded.
-
-Also note that developer adoption of the `beforematch` event in these use
-cases should be straightforward, since the typical page that provides content in
-collapsed sections already has an event handler to expand and show the section.
-With the `beforematch` event, we can reuse the same handler to expand the
-section.
+wikipedia pages. find-in-page and scroll-to-text currently can't find text
+inside of the collapsed sections, but with `hidden=until-found` they could be
+searchable and automatically expanded.
 
 Using this feature is a good optimization for low power mobile devices that
 don't want to render all of the content in the page but still making it
 accessible to features such as find-in-page. Making long articles with collapsed
 sections accessible to find-in-page will improve the user experience.
 
+## The `beforematch` event and other use cases
+
+The previous example doesn't need to use the proposed `beforematch` event
+because all other visual state in the example is controlled with HTML attribute
```

Comment:
I suspect we'll have to update this.

There are sibling combinators, but they only match the second sibling, so I don't know if you can do this with css alone without some awkward positioning

Reply:
Are you saying that we should add a new example that actually uses `beforematch`?

Reply:
Now that I've actually tried it I see.
I updated it, and now it also uses beforematch.

I also removed the section which says that the example doesn't use beforematch and justifies why we need beforematch.